### PR TITLE
Adding Controller Framework that is used by Controller and Provisione…

### PR DESCRIPTION
**Pull request by brahmaroutu**
…r as common code

Adding controller basic code that is used by other  packages/repos. 

assign @wlan0

### CODE REVIEW

**Comment by wlan0** [comment]
The post build step for the previous commit failed: https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/post-container-object-storage-interface-api-push-images/1330951515501760512

```
2020/11/23 19:09:08 /home/prow/go/src/github.com/kubernetes-sigs/container-object-storage-interface-api/cloudbuild.yaml does not exist
```

Could you look into it please

File: controller/controller.go [hunk 1]
```diff
@@ -0,0 +1,520 @@
...
+	default:
+		panic("unknown item in queue")
+	}
+
+	lockKey := fmt.Sprintf("%s/%s", key, ext)
+	if c.locker == nil {
+		c.locker = map[string]*sync.Mutex{}
+	}
+	if _, ok := c.locker[lockKey]; !ok {
+		c.lockerLock.Lock()
```
Comment (wlan0) [inline controller/controller.go:313]: We need to acquire this lock outside the if-condition. When there are multiple requests for the same key (that does not exist yet), they could race and lead to concurrent map access panic. 


This should be outside the if-block
```
c.lockerLock.Lock()
defer c.lockerLock.Unlock()
```

Comment (brahmaroutu) [reply controller/controller.go]: Fixed

**Comment by brahmaroutu** [comment]
@wlan the post trigger logic will be only for builds that need to push built images, this repo does not need one.

**Pull request merged**